src/index.test.ts

```diff
@@ -0,0 +1,97 @@
+import { describe, test, beforeEach } from "node:test";
+import assert from "node:assert/strict";
+import { promisify } from "util";
+import { exec as lameExec } from "child_process";
+const exec = promisify(lameExec);
+import { Keypair } from "@solana/web3.js";
+import { getKeypairFromFile, getKeypairFromEnvironment } from "./index";
+import { writeFile } from "node:fs/promises";
+import base58 from "bs58";
+
+const log = console.log;
+
+const TEMP_DIR = "src/temp";
+
+describe("getKeypairFromFile", () => {
+  test("getting a keypair from a file", async () => {
+    const TEST_FILE_NAME = `${TEMP_DIR}/test-keyfile-do-not-use.json`;
+    const { stdout, stderr } = await exec(
+      `solana-keygen new --force --no-bip39-passphrase -o ${TEST_FILE_NAME}`,
+    );
+    assert(stdout.includes("Wrote new keypair"));
+    const keypair = await getKeypairFromFile(TEST_FILE_NAME);
+  });
+
+  test("throws a nice error if the file is missing", async () => {
+    try {
+      await getKeypairFromFile("I DONT EXIST");
+    } catch (thrownObject) {
+      const error = thrownObject as Error;
+      assert.strictEqual(
+        error.message,
+        `Could not read keypair from file at 'I DONT EXIST'`,
+      );
+    }
+  });
+
+  test("throws a nice error if the file is corrupt", async () => {
+    const TEST_FILE_NAME = `${TEMP_DIR}/corrupt-keyfile-do-not-use.json`;
+    await writeFile(TEST_FILE_NAME, "I AM CORRUPT");
+    try {
+      await getKeypairFromFile(TEST_FILE_NAME);
+    } catch (thrownObject) {
+      const error = thrownObject as Error;
+      assert.strictEqual(
+        error.message,
+        `Invalid secret key file at '${TEST_FILE_NAME}'!`,
+      );
+    }
+  });
```

Comment:
Is it perhaps preferable to have _known_ fixtures from which you read, instead of creating them on the fly? When you create them on the fly you're testing _three_ things (the workability of your filesystem, `solana-keygen`, and your actual code).

Reply:
Done! Moved it all to before()